crates/ruff/src/jupyter/notebook.rs

```diff
@@ -37,7 +29,53 @@ pub fn is_jupyter_notebook(path: &Path) -> bool {
         && cfg!(feature = "jupyter_notebook")
 }
 
-impl JupyterNotebook {
+/// Return `true` if the given [`Cell`] is a valid code cell.
+///
+/// A valid code cell is a cell where the type is [`CellType::Code`] and the
+/// source is not a magic, shell or help command.
+fn is_valid_code_cell(cell: &Cell) -> bool {
+    if cell.cell_type != CellType::Code {
+        return false;
+    }
+    let lines = match &cell.source {
+        SourceValue::String(string) => string.lines().collect::<Vec<_>>(),
+        SourceValue::StringArray(string_array) => string_array
+            .iter()
+            .map(std::string::String::as_str)
+            .collect(),
```

Comment:
Correct me if I'm wrong, but that's a vector of string slice (`&str`), so is it not just collecting into a vector of view to the actual `String` and _not_ copying it? The only reason I'm collecting is to avoid code repetition, although one could argue against it :)

Reply:
Right, I missed that. I think it could still make sense to move the `any` into the match arms to avoid the unnecessary allocation (and writing a vec entry for every cell).